### ISSUE TITLE
Allow Validators to Limit Maximum Amount of Atoms Delegated

### ISSUE THREAD

**Issue by crainbf** [cosmos/cosmos-economics#28]
There has been some discussion about this previously, but no issue so far. 

I think validators should be able to limit how many atoms are delegated to them. This should ideally be expressed as a percentage of the total atom supply (or possibly of the amount of bonded atoms). 

The reason is to give a tool to responsible validators to prevent them from becoming too big. Otherwise, high quality / well-known validators may have no way to prevent more and more atoms to be delegated to them. 

Thus, for example, a validator could choose that never more than 10% of atoms can be staked with their pool. Then once that threshold is reached, no more delegation transactions can take place.

This can also provide a tool for the atom community to get a wider distribution among validators. For example, a standard could emerge where atom holders put pressure on validators to impose such a constraint on themselves. Or an atom holder could use that as a criterion in selecting a validator. e.g. I won't delegate to a validator who doesn't restrict size to less than 12%.

There are some open questions for me on how to specify this precisely. For example:

- Since the atom supply grows over time, the cap can't be expressed as a fixed number. This would entail the percentage shrinking over time, which doesn't make sense.
- There should be some way for validators to adjust the cap. However, this is obviously prone to abuse so it needs to be severely restricted or delayed. I could imagine one option being that a validator can make a set a new cap, but it only becomes active with a 2-month delay. This way, atom holders can use social media/etc to coordinate and move away from that validator in time.
- I don't think atoms should be unbonded if a validator goes above the limit for reasons outside their control. e.g. let's say a validator has a limit of 10% and then some other validator gets slashed, so all of a sudden she is at 12%. In that case, it shouldn't be possible to delegate new atoms to them but no unbonding should be happen.

Probably, it makes more sense to express the cap as a percentage of the amount of bonded atoms. In that case, too, I don't think automatic unbonding should happen if the #of bonded atoms decreases and thus the cap is exceeded.
So let's say. A says cap of 10% when 70% of atoms are bonded. So 7% of total atom supply and is currently at the maximum. Now the amount of bonded atoms decreases to 65%, so now the validator now controls 7/65=10.8% of bonded atoms. It shouldn't be possible to delegate new atoms to them until the validator falls below the threshold.

In my view, such a cap could be a powerful tool to prevent situations like we had in Bitcoin when some mining pools got to huge amounts of hash rate before.

**Comment by rigelrozanski** [comment]
I'm wondering if there are any other existing mechanisms in place which could be used to achieve the same effect? Commission rate would be one, however we have added a self imposed maximum... so for starters, a validator could increase their commission to their maximum, that should shed them of some delegators. 
Another mechanism would be simply alerts within the UI for the delegators - we were thinking about this one as a suggestion mechanism to discourage folks from centralizing delegations. 

> Otherwise, high quality / well-known validators may have no way to prevent more and more atoms to be delegated to them.

Isn't that the point that good validators get more atoms delegated to them however? I suppose there is the factor of adding unwanted additional heat... but maybe we don't want validators who can't adapt to the needs of the network.. (maybe? just saying).

However I see you points and I think I'm generally am in favour. From a technical standpoint, I think all of technical points you raised are valid and you've addressed them correctly

**Comment by crainbf** [comment]
Sure, commission rate increasing is an option. But I don't like it for several reasons:
First, let's say as a validator we have customers and they are paying 20%, which seems fair to both sides. Then more and more people delegate, but we don't want to grow too big. Raising prices on everyone until you piss off enough people until they leave just feels very ugly to me. I really wouldn't want to do that.

Second, up to a point a good validator should get more delegators. But for example, I wouldn't want to go above a certain threshold. Maybe 15% or at the very maximum 20%, because beyond that it is bad for the network and just accumulates too much power in the hand of that validator. So as a validator interested in the long term success of the network, I'm interested in having a diversity of validators and thus wouldn't want to become too big.

**Comment by jaekwon** [comment]
We can cap the amount of staking power any single validator (with delegation) can get at say 10%.  But I don't think it makes sense to otherwise limit delegation, because some people will want to invest-in/bootstrap validators by delegating to them.

**Comment by jaekwon** [comment]
I think what you want is for there to be some global limit to how big a validator (including delegated voting power) can get.  That addresses the problem directly.  How about say 10%?

Of course anyone can create more identities, but I think that's fine if they're actually decentralizing and running more nodes.

**Comment by crainbf** [comment]
I think there are many reasons, why I prefer the approach I outlined above. 

For example:
1) If we pick a global limit that figure is necessarily arbitrary. However, if we allow validators to set their own limit and delegators to use that as a criterion to select who to delegate we assign this as a responsibility to the community and give them a powerful tool to find the right balance.

2) I also don't think that one global limit makes sense, since validators will have very different setups. For example: 
- how quickly a validator gets unbonded if she doesn't sign any blocks depends on her size. So it may make sense for a validator with a super robust setup who is confident that they can maintain very high availability to choose a limit of say 12%. On the other hand, a validator who may not have around-the-clock support staff, etc may want to choose a much lower limit to prevent getting unboded. 
- Another example is that validators should choose different limits simply based on how confident they are in their security setup. A new validator could choose a low limit until they are fully confident of its security and then increase it over time. (Again, let's say with that 2-month delay to give delegators a chance to switch if they want to.)

3) Finally, I think us setting the limit sends the wrong signal. We are indicating to the community that we see it as our task and our ability to prevent centralization among validators. But I don't think it should be our task, but the task of the delegators to ensure that. I also don't think that a global limit is a very effective tool to prevent this risk. A validator splitting up over several keys is not a great solution. They'd likely replicate the exact same setup. I feel above outline will clearly communicate that it is the tasks of the delegators (and validators) to ensure a balanced distribution of power.

**Comment by rigelrozanski** [comment]
I like your points Brian, and I'm still in favour of this but I'm going to make a contrary points... 
Note that it's not AiB, but cosmos-governance which would get to ultimately choose what that global maximum should be...  and it's simple. We're going for the absolute simplest solution which works reasonably well, so I think the question is how much not including this parameter will deter good validators from joining the system.

**Comment by crainbf** [comment]
Of course, in either way, governance could always change it. Either by removing individual caps and replacing them with a global cap or the other way around. 
That being said, my feeling is that it will not be so easy to go from global cap to validator set cap. It's not just a technical thing, but also a question of social norms. If delegators should take into account a validator's self-imposed cap, it should be there in the beginning so people get used to it.

Your point about simplicity is, of course, valid. Do you think it would be complex/difficult to implement above?

I don't necessarily think that this point would prevent good validators from becoming validators in the first place, but I do think it would lead to a better and more responsible community.

(With all that being said, if the choice is between no cap and a global cap, I would prefer to have a global cap. Maybe 15% or so. However, I would much, much prefer validator-set caps.)

**Comment by jaekwon** [comment]
Ok, all good points.  Lets see what @sunnya97 thinks too.

**Comment by rigelrozanski** [comment]
No it's not too complicated to implement - this is relatively straight forward minus a few special circumstances which you previously outlined. Any complexity would be on the delegators side, having to understand this parameter - but that can maybe be solved with UI

**Comment by crainbf** [comment]
@rigelrozanski Awesome! Thanks.

Yes, it would be great to get Sunny and Zaki's input too.

**Comment by sunnya97** [comment]
I agree with Brian's points.  Only possible change I might suggest, is that perhaps validators shouldn't be able to set their max delegation percentage below their percentage of self-bond.  That way, the sum of all the validators max delegation percentage is greater than 100%, thus it isn't possible for the validators to prevent new delegators from partaking in staking.


>Of course anyone can create more identities, but I think that's fine if they're actually decentralizing and running more nodes.
Yeah, not a fan of the global cap. We don't want to incentivize them making more validators, because they're not *really* more validators.  It's one validator that's gonna be signing blocks with two different keys.  Also, especially because we have a cap on validators, having single entities make multiple validators is undesirable because it pushes out someone from the bottom, making there be less actual validators.

**Comment by ebuchman** [comment]
What if all the validators set caps and then theres all these atoms that just can't delegate because everyone is at cap? 

I'm also concerned about the additional complexity of supporting these extra rules.

I believe we were working on this in October in the form of lower rewards if too much was delegated, rather than a hard cap.

**Comment by crainbf** [comment]
@ebuchman I would expect most validators to set fairly high caps (e.g. 10% to 20%). So I think the likely outcome is that we'd get an aggregate of caps far above 100%. I don't think the scenario where lots of atoms can't delegate is plausible / a stable state. After all, that means a new validator can come along, and allow all these people to delegate to them. Or existing validators have an incentive to increase their caps. 

I understand the concern with complexity, but I really think this is a crucial feature to have.

I don't think decreasing rewards when more atoms are being delegated is a good way of achieving these things. 

(Happy to elaborate on that if desired.)

**Comment by ebuchman** [comment]
Curious to understand why caps are better than lower rewards.

**Comment by sunnya97** [comment]
> What if all the validators set caps and then theres all these atoms that just can't delegate because everyone is at cap?

@ebuchman the thing I suggested solves that.  

> validators shouldn't be able to set their max delegation percentage below their percentage of self-bond
